Stripe/STPAPIClient.m

```diff
@@ -356,24 +357,33 @@ + (BOOL)canSubmitPaymentRequest:(PKPaymentRequest *)paymentRequest {
     return [[[paymentRequest.paymentSummaryItems lastObject] amount] floatValue] > 0;
 }
 
-+ (NSArray<NSString *> *)supportedPKPaymentNetworks {
++ (NSArray<NSString *> *)supportedPaymentNetworksForCountry:(NSString *)countryCode {
     NSArray *supportedNetworks = @[PKPaymentNetworkAmex, PKPaymentNetworkMasterCard, PKPaymentNetworkVisa];
-    if ((&PKPaymentNetworkDiscover) != NULL) {
+    if ((&PKPaymentNetworkDiscover) != NULL && [countryCode isEqualToString:@"US"]) {
         supportedNetworks = [supportedNetworks arrayByAddingObject:PKPaymentNetworkDiscover];
     }
     return supportedNetworks;
 }
 
 + (BOOL)deviceSupportsApplePay {
-    return [PKPaymentAuthorizationViewController class] && [PKPaymentAuthorizationViewController canMakePaymentsUsingNetworks:[self supportedPKPaymentNetworks]];
+    return [self deviceSupportsApplePayInCountry:@"US"];
+}
+
++ (BOOL)deviceSupportsApplePayInCountry:(NSString *)countryCode {
+    NSArray<NSString *> *supportedNetworks = [self supportedPaymentNetworksForCountry:countryCode];
+    return [PKPaymentAuthorizationViewController class] && [PKPaymentAuthorizationViewController canMakePaymentsUsingNetworks:supportedNetworks];
 }
 
 + (PKPaymentRequest *)paymentRequestWithMerchantIdentifier:(NSString *)merchantIdentifier {
+    return [self paymentRequestWithMerchantIdentifier:merchantIdentifier country:@"US"];
+}
+
++ (PKPaymentRequest *)paymentRequestWithMerchantIdentifier:(NSString *)merchantIdentifier country:(NSString *)countryCode {
     PKPaymentRequest *paymentRequest = [PKPaymentRequest new];
     [paymentRequest setMerchantIdentifier:merchantIdentifier];
-    [paymentRequest setSupportedNetworks:[self supportedPKPaymentNetworks]];
+    [paymentRequest setSupportedNetworks:[self supportedPaymentNetworksForCountry:countryCode]];
     [paymentRequest setMerchantCapabilities:PKMerchantCapability3DS];
-    [paymentRequest setCountryCode:@"US"];
+    [paymentRequest setCountryCode:countryCode];
     [paymentRequest setCurrencyCode:@"USD"];
```

Comment:
I think we should consider not defaulting currency to USD and force people to specify.